src/Text.tsx

```diff
@@ -1,13 +1,53 @@
+import React from 'react'
 import styled from 'styled-components'
 import {COMMON, SystemCommonProps, SystemTypographyProps, TYPOGRAPHY} from './constants'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from './utils/polymorphic'
 import sx, {SxProp} from './sx'
-import {ComponentProps} from './utils/types'
 
-const Text = styled.span<SystemTypographyProps & SystemCommonProps & SxProp>`
+const restrictedAs = [
```

Comment:
`restrictedAs` might be a confusing name for this variable because it could imply that the following values are restricted. Maybe `allowedElements` is more clear?